pypsa/linopf.py

```diff
@@ -1194,6 +1233,7 @@ def prepare_lopf(
     # consider only state_of_charge_set for the moment
     define_fixed_variable_constraints(n, snapshots, "StorageUnit", "state_of_charge")
     define_fixed_variable_constraints(n, snapshots, "Store", "e")
+    define_modular_nominal(n, c="Generator", attr="n_opt", attr_nom="p_nom")
```

Comment:
For Fabians, this is easy to generalize to all components and it is a next step, however, we would like to have a preliminary check to avoid overhead in the next stages